src/ui/public/visualize/loader/visualize_data_loader.ts

```diff
@@ -73,10 +73,15 @@ export class VisualizeDataLoader {
     this.vis.showRequestError = false;
 
     try {
+      const minimalColumns =
```

Comment:
Could we maybe add a short comment here, why we calculate `minimalColumns` like that, i.e. what are the use-cases we are trying to check for?